ado-agents.tf

```diff
@@ -0,0 +1,215 @@
+resource "kubernetes_namespace" "ado-agents_namespace" {
+  count = var.ado-agents_config.enable ? 1 : 0
+  metadata {
+    name = var.ado-agents_config.namespace
+    labels = {
+      "app.kubernetes.io/managed-by" = "Terraform"
+      "istio-injection"              = "disabled"
+    }
+  }
+  depends_on = [time_sleep.wait_for_aks_api_dns_propagation]
+}
+
+
+resource "kubernetes_secret" "azdevops" {
+  count = var.ado-agents_config.enable ? 1 : 0
+  metadata {
+    name      = var.ado-agents_config.secretname
+    namespace = var.ado-agents_config.namespace
+  }
+
+  data = {
+    "${var.ado-agents_config.secretkey}" = var.ado-pat
+  }
+
+  type = "Opaque"
+  depends_on = [
+    kubernetes_namespace.ado-agents_namespace
+  ]
+}
+
+# var.ado-agents_config.enable true AND var.ado-agents_config.scaledjob set to False so Install Job that is also requires KEDA
+resource "kubernetes_job" "azdevops_agent" {
```

Comment:
what is this for ?

Reply:
So when I read ticket description it was mentioning Job only and example so this implements Job, But after that I also implemented ScaledJob (which actually uses KEDA). On the example on internet modules implements this actually both implements Job and ScaledJob https://github.com/tonyskidmore/terraform-kubernetes-azure-devops-agent/blob/main/helm/charts/job/templates/ so both implemented but with vars you can choose between Job or ScaledJob:

ado-agents_config = {
  enable = true
  # So we use KEDA ScaledJob not Job
  scaledjob  = true

Reply:
We don't want job , we want only scaled job